idb.js

```diff
@@ -7,13 +12,16 @@ const dbp = new Promise((resolve, reject) => {
 
 export const call = async (type, method, ...args) => {
```

Comment:
we could have an early return here directly, shouldn't we throw if not supported ?
```js
if (!hasIDB) return reject(Error('indexeDB not found'))
```

Reply:
mmm just starting to test this now, although I suspect throwing might lead to SSR failing in practice. I think from a practical point of view if index DB doesn't exist the user should expect just to return the default value. anyway I'll test that assumption now that rejecting will case SSR builds to fail in my codebase at least anyway.

Reply:
yep just tried this change and it crashes my SSR code, and build code in a gatsby project.

in SSR we if hasIDB is false then it should gracefully degrade to the returning the default value which will make the code SSR safe.

seeing I technically haven't made your requested change I won't hit the resolve button so that you can review it :)

Reply:
well the idea would be to catch it in your usercode and handle it the way you want.

Reply:
Interesting, before this PR my code base would crash even just from importing it (unless you assigned it with a conditional require statement).

So... We can leave error checking to user but needs to at least not fail on import. I'll double check that's working in my code base later tonight for you. :)

Reply:
ok just to confirm if we reject here because this promise runs straight away on import it will cause any SSR code to error just by importing it. (unless you instead assign it with a conditional require statement)

```javascript
import { useIdb } from 'react-use-idb'   /// this will crash SSR code

// alternatively you could use a conditional require statement
const useIdb = typeof window !== 'undefined' ? require('react-use-idb') : null;
```

@kigiri which Option Do you want me to run with to Resolve?
1. (assuming you prefer to pass error to user code then
  - delay execution of that promise to when the main method is called?
  - or any other ideas?
2. (alternatively)
  - gracefully degrade to default if window or indexedDB isn't defined

I'll keep playing to see if another way :)